src/server/game/Entities/Object/Object.cpp

```diff
@@ -2539,6 +2612,69 @@ float WorldObject::GetFloorZ() const
     return std::max<float>(m_staticFloorZ, GetMap()->GetGameObjectFloor(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZ()));
 }
 
+void WorldObject::SetTransport(Transport* transport)
+{ 
+    if (transport) 
+    {
+        m_movementInfo.transport.guid = transport->GetGUID();
+        // could it be possible to initialize the rest of the transport data (seat, time, offset) here?
+        AddUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+    else // if(!m_vehicle) ?
+    {
+        m_movementInfo.transport.Reset();
+        RemoveUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+
+    m_transport = transport;
+}
+
+MovementInfo WorldObject::GetMovementInfo() const
```

Comment:
Why is this function building a fresh MovementInfo object every time?

Reply:
Also the returned object may contain a different position than the ones contained in m_movementInfo because we use the positions from the object Position inheritance instead.